### ISSUE TITLE
adapt code for pytorch 1.2

### ISSUE THREAD

**Issue by harbecke** [harbecke/HexHex#25]
[pytorch 1.2](https://pytorch.org/blog/pytorch-1.2-and-domain-api-release/) was released. some of the relevant features for us are:
- tensorboard is natively supported: `from torch.utils.tensorboard import SummaryWriter`
- AdamW optimizer is included (an Adam Optimizer that works with weight decay)
- there is a `nn.Transformer` module, if we want to experiment with transformer models
